frontend/src/scenes/data-warehouse/editor/QueryWindow.tsx

```diff
@@ -143,8 +152,85 @@ export function QueryWindow({ onSetMonacoAndEditor }: QueryWindowProps): JSX.Ele
+                                    )
+                                } else {
+                                    saveOrUpdateDraft(
+                                        {
+                                            kind: NodeKind.HogQLQuery,
+                                            query: queryInput,
+                                        },
+                                        undefined,
+                                        currentDraft?.id || undefined,
+                                        undefined
+                                    )
+                                }
+                            }}
+                        >
+                            Save
+                        </LemonButton>
+                        <LemonButton
+                            type="tertiary"
+                            size="xsmall"
+                            id="sql-editor-query-window-publish-draft"
+                            disabledReason={editingViewDisabledReason}
+                            onClick={() => {
+                                if (editingView && currentDraft?.id) {
+                                    publishDraft(currentDraft.id, {
+                                        id: editingView.id,
+                                        query: {
+                                            ...sourceQuery.source,
+                                            query: queryInput,
+                                        },
+                                        name: editingView.name,
+                                        types: response && 'types' in response ? response?.types ?? [] : [],
+                                        shouldRematerialize: isMaterializedView,
+                                        edited_history_id: currentDraft.edited_history_id,
+                                    })
+                                } else {
+                                    saveAsView(false, currentDraft?.id)
+                                }
+                            }}
+                            tooltip={
+                                !editingView
+                                    ? 'The view this draft is based on has been deleted. Publishing will create a new view.'
+                                    : ''
+                            }
+                        >
+                            {!editingView && <IconInfo className="mr-1" color="var(--warning)" />}
+                            Publish
+                        </LemonButton>
+                    </>
+                )}
+                {editingView && !isDraft && (
                     <>
+                        <LemonButton
+                            type="tertiary"
+                            size="xsmall"
+                            id="sql-editor-query-window-save-draft"
+                            onClick={() => {
+                                if (activeModelUri) {
```

Comment:
It may be worth moving this up to the above boolean `{editingView && !isDraft && activeModelUri && (` so that this `onClick` is more stable